### DISCUSSION TITLE
should jupyterlab plotly backend work with retrolab?

### DISCUSSION THREAD

**Discussion by fdion** [jupyterlab/retrolab#320] [Q&A]
Given the following code (minimal working example):

```
import plotly.express as px
fig = px.scatter_3d(px.data.iris(), x='sepal_length', y='sepal_width', z='petal_length', color='species', size='petal_width')
fig.show()
```

I get output in `jupyterlab` and in classic `notebook`, but nothing in `retrolab`. returning the figure directly, using fig or fig.show() makes no difference.

If I modify the code as:
```
import plotly.express as px
import plotly.io as pio
pio.renderers.default = 'iframe'

fig = px.scatter_3d(px.data.iris(), x='sepal_length', y='sepal_width', z='petal_length', color='species', size='petal_width')
fig.show()
```

I do get some output. `iframe` is the only renderer that seems to work. I am aware of #258 (and of #262) and rolled back to 0.3.12 of retrolab, but I get the same behaviour. **Which backend should be used**? Once that is clear, I'll open an issue with plotly if that is where the problem lies.

**Comment by jtpio** [comment]
Thanks @fdion for reporting.

I just tested on Binder with this gist, which uses the latest `0.3.15` release of RetroLab: https://gist.github.com/jtpio/4a6a34da66b32249e28df718ec30e4d7

And the figure is displayed in all the interfaces:

https://user-images.githubusercontent.com/591645/148424540-f362b218-ce34-4fda-806c-1dcc28211b61.mp4

Maybe updating to the latest `0.3.15` would help?

**Reply by fdion** [reply]
I first started with `0.3.15` and switched to `0.3.12` since it didn't work. If I create an environment with python 3.9, then add pandas, plotly and retrolab only, it does work.

Will need to dig deeper as what makes the full environment doesn't work. Thanks @jtpio

**Reply by fdion** [reply]
So far, this only happens on Windows, so not a huge deal as I use Linux. And it only happens if I set up an environment for jupyter retro, with individual environment / kernels.

**Reply by bollwyvl** [reply]
>  up an environment for jupyter retro, with individual environment / kernels.

This might be the thing: many labextension packages co-locate the backend and frontend components to keep them in sync, so the given package might need to be installed in both for a visualization to appear.
